### PR TITLE
feat(snapshot): enforce json format.

### PR DESCRIPTION
**Pull request by louis-bompart**
## Proposed changes

To ensures consistent file formating, let's enforce the format when pulling, previewing etc snapshots.

## Testing

- [X] Unit Tests: done
- [x] Manual Tests: Checked Expanded Preview, there are no more space changes in the commits

-----
CDX-491

### CODE REVIEW

File: packages/cli/src/lib/project/project.ts [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 import {cli} from 'cli-ux';
 import archiver from 'archiver';
 import {InvalidProjectError} from '../errors';
 import extract from 'extract-zip';
 import {DotFolder, DotFolderConfig} from './dotFolder';
+import {readJsonSync, writeJsonSync, WriteOptions} from 'fs-extra';
 
 export class Project {
   private static readonly resourceFolderName = 'resources';
+  public static readonly jsonFormat: WriteOptions = {spaces: '\t'};
```
Comment (y-lakhdar) [inline packages/cli/src/lib/project/project.ts:18]: image

Comment (y-lakhdar) [reply packages/cli/src/lib/project/project.ts:18]: I don't think this will be enough to enforce a JSON format

**Review by y-lakhdar — COMMENTED** [review]
Instead of replacing spaces with tabs, can't we just lint the file after pulling it?
It would be nice to add a linter in the Coveo project. 

I always format my JSON files (with the VScode extension) when I manipulate them. I'm sure I am not the only one who does that.

Here is a JSON file returned by the API and the formatted version (using the VScode JSON formatter)
image

File: packages/cli/src/lib/project/project.ts (hunk 1 above)
Comment (louis-bompart) [reply packages/cli/src/lib/project/project.ts:18]: Why so, can you elaborate?

Comment (y-lakhdar) [reply packages/cli/src/lib/project/project.ts:18]: As shown in the screenshot above, the JSONs (the one from the project and the other coming from the API) could have more than just space and tabs differences.

Comment (louis-bompart) [reply packages/cli/src/lib/project/project.ts:18]: Hm 👍 to keep a trace of what we discuss, I think it's not a problem 'cause what is the true goal here is to enforce formating in the expanded preview, which should be done here I reckon

**Review by y-lakhdar — APPROVED** [review]

**Pull request merged**